src/util.c

```diff
@@ -40,17 +40,36 @@
  */
 char *xmpp_strdup(const xmpp_ctx_t *ctx, const char *s)
 {
-    size_t len;
+    return xmpp_strndup(ctx, s, SIZE_MAX);
+}
+
+/** Duplicate a string with a maximum length.
+ *  This function replaces the standard strndup library call with a version
+ *  that uses the Strophe context object's allocator.
+ *
+ *  @param ctx a Strophe context object
+ *  @param s a string
+ *  @param len the maximum length of the string to copy
+ *
+ *  @return a new allocates string with the same data as s or NULL on error
```

Comment:
Please, change this line to
```
a newly allocated string that contains at most `len` symbols of the original string or NULL on error
```

Reply:
hmm, I wouldn't use the term 'symbols' here as in reality it's 'bytes' ... (as also used in the man pages for strndup)